src/main/scala/gen/gmos/GmosNLongslit.scala

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic.gen.gmos
+
+import basic.ObservingMode
+import basic.search.Constraints
+import basic.syntax.all._
+
+import gem.Step
+import gem.config.DynamicConfig.GmosN
+import gem.config.GmosConfig._
+import gem.enum._
+import gem.math.Wavelength
+
+import fs2.{Pure, Stream}
+
+import java.time.Duration
+
+/**
+ * Sequence generation for GMOS North Longslit.
+ */
+sealed trait GmosNLongslit {
+
+  /**
+   * Produces an infinite acquisition sequence. The idea is that the sequence
+   * should be stopped when the target is acquired.
+   */
+  def acquisition(e: Duration): Stream[Pure, Step.GmosN]
+
+  /**
+   * Produces an infinite science sequence.  Each sequence element is a finite
+   * mini-sequence containing steps that must be executed together. The
+   * sequence should be stopped when the desired S/N ratio has been met.
+   */
+  def science(e: Duration): Stream[Pure, Stream[Pure, Step.GmosN]]
+
+}
```

Comment:
I don't really know what we should use for the return values.  There are no effects, assuming the exposure time is passed in as an argument having been calculated from ITC.  I wanted it to be an infinite stream but it could just as easily have been a plain `scala.collection.immutable.Stream` I suppose.  The nested `Stream` in the science sequence was chosen (instead of say a `Stream` of `List` or `Array`) to make it simple to just `.flatten`.

Reply:
If we want infinite streams I'm happier with `fs2.Stream` because of the memory hazards of stdlib `Stream`. We can always re-evaluate later if it doesn't need to be this general.